Source/Extensions/Blazorise.TreeView/TreeView.razor.cs

```diff
@@ -7,19 +7,19 @@
 
 namespace Blazorise.TreeView
 {
-    public partial class TreeView<TNode> : BaseComponent
+    public partial class TreeView : BaseComponent

```

Comment:
Changing from generic to the non-generic component is a breaking change. That means if we did that, it would be possible only in v2.0.

The alternative would be to have an internal `TreeNode<TNode>` that would hold the node state. That way TreeView API would not change.

Reply:
Well, I think `AllExpanded` can also work, as long as we keep track of internal changes in `SetParametersAsync`. But having both ways is also a good approach. Just like we have the `Visible` parameter and `Show`/`Hide `methods on some other components.

Reply:
I am wondering how can we access `IsExpanded` internally.  For need, this property because of collapse/expand for each node. I was thinking about the below code but it looks like the partial class doesn't support where clause.  

```csharp
public interface ITreeNode
{
        bool IsExpanded { get; set; }
}

public partial class _TreeViewNode<TNode> : BaseComponent
        where TNode:  ITreeNode
{
}

Reply:
> Yes, we ask that you focus on the PR itself and what it implements and not changing the public facing API at all, these should not be removed. ![image](https://user-images.githubusercontent.com/22283161/191328912-83a38658-b9e6-4cfe-b124-b57337025bf4.png) If you have suggestions about the public facing API we can consider that for the next major version **v2.0** but on a separate PR please. Thanks!
> 
> Also I'd argue a **ExpandAll** method would be more flexible then a flag **AllExpanded** that only works on `Initialized`? What do you guys think?

I agree, I accidentally remove `[Parameter] public EventCallback<TNode> SelectedNodeChanged { get; set; }` this is not related to this PR.

Reply:
> I am wondering how can we access `IsExpanded` internally. we need this property because of collapse/expand for each node. I was thinking about the below code but it looks like the partial class doesn't support the where clause.
> 
> ```cs
> public interface ITreeNode
> {
>         bool IsExpanded { get; set; }
> }
> 
> public partial class _TreeViewNode<TNode> : BaseComponent
>         where TNode:  ITreeNode
> {
> }
> ```

Also, in this case, we will have breaking changes. Any ideas?

Reply:
Hmmm I can't recall the Tree View source code now and I'm on mobile now. But I think you'll have to either check the expanded nodes collection for the reference so you can tell whether it's expanded.

Or keep internal state of the isexpanded some other way, which I believe there must be a way to track.

I guess it's weird that the expanded nodes are tracked by a collection instead of the isexpanded state in each node.